swarm/swarm.go

```diff
@@ -213,9 +226,29 @@ func (self *Swarm) Start(srv *p2p.Server) error {
 		}
 	}
 
+	go self.metricsLoop()
+
+	startCounter.Inc(1)
 	return nil
 }
 
+// metricsLoop periodically sends metrics about storage
+func (self *Swarm) metricsLoop() {
+	ticker := time.NewTicker(metricsTimeout)
+
+	go func() {
+		for _ = range ticker.C {
+			self.sendMetrics()
```

Comment:
This loop should exit if the node is stopped.

Reply:
I chose a "lazy" solution to this as I concluded we'll be collecting those metrics for the complete duration of a swarm process run. As far as I understand, a swarm node `stop` means somehow the shutdown of the process, in which case it would be somehow killed, killing also this loop. This may be a bad assumption, in which case I will add an exit path.